src/js/form.js

```diff
@@ -383,9 +440,12 @@ Form.prototype.getModelValue = function( $control ) {
 /**
  * Finds nodes that have attributes with XPath expressions that refer to particular XML elements.
  *
- * @param  {string} attr - The attribute name to search for
- * @param  {?string} filter - The optional filter to append to each selector
- * @param  {{nodes:Array<string>=, repeatPath: string=, repeatIndex: number=}=} updated - The object containing info on updated data nodes
+ * @param {string} attr - The attribute name to search for
+ * @param {string} [filter] - The optional filter to append to each selector
+ * @param {object} [updated] - The object containing info on updated data nodes
+ * @param {Array<string>} [updated.nodes]
+ * @param {string} [updated.repeatPath]
+ * @param {number} [updated.repeatIndex]
```

Comment:
I think it might be good to define the updated object centrally, since this is such a key part of the engine, and occurs many times (calculate.js, itemset.js, relevant.js, repeat.js, required.js, output.js, readonly.js). What do you think?

Reply:
You're right. I created the `@typedef` for `UpdatedDataNodes`